src/headerssync.h

```diff
@@ -215,7 +215,7 @@ class HeadersSyncState {
 
     /** In REDOWNLOAD, check a header's commitment (if applicable) and add to
      * buffer for later processing */
-    bool ValidateAndStoreRedownloadedHeader(const CBlockHeader& header);
+    bool ValidateAndStoreRedownloadedHeader(const CBlockHeader& header, bool from_cache);
```

Comment:
as stated before, we likely need to rename the method now, since we're not "downloading" from a cache anymore